autoload/test/csharp/xunit.vim

```diff
@@ -0,0 +1,73 @@
+if !exists('g:test#csharp#xunit#file_pattern')
+  let g:test#csharp#xunit#file_pattern = '\v\.cs$'
+endif
+
+function! test#csharp#xunit#test_file(file) abort
+  if fnamemodify(a:file, ':t') =~# g:test#csharp#xunit#file_pattern
+    if exists('g:test#csharp#runner')
+      return g:test#csharp#runner ==# 'xunit'
+    else
+      let l:project_path = test#csharp#get_project_path(a:file)
+      let l:dotnet_xunit_cli = filereadable(l:project_path) 
+          \ && match(readfile(l:project_path), 'DotNetCliToolReference.*dotnet-xunit')
+      echom l:project_path
+      return l:dotnet_xunit_cli 
+          \ && (search('using Xunit', 'n') > 0)
+          \ && (search('[(Fact|Theory).*]', 'n') > 0)
+    endif
+  endif
+endfunction
+
+function! test#csharp#xunit#build_position(type, position) abort
+  " TODO: if we can hook into the before_run function, then in theory we
+  " could set the project_root dynamically before each run, which would
+  " remove the necessity of setting the g:test#project_root manually.
+  "
+  "let file = a:position['file']
+  "let project_path = test#csharp#get_project_path(file)
+  "let g:test#project_root = fnamemodify(project_path, ':p:h')
```

Comment:
Does this commented out code need to remain?

Reply:
Not necessarily. At the moment there is a bug, that the pwd needs to be the same as the solution file and if you have sub-folders in the project and autochdir enabled, then it will not work unless you manually change the folder to the project root. I thought that setting it dynamically would work, but that does not work and it needs the changes that are introduced in #242 or something similar.

Shall I remove this commented out code and create an issue?

Reply:
Yes I think that would be better as commented out code can be a source of confusion. If you do create an issue you can add in the link in the comment block.